tests/system/framework/roles/shadow.py

```diff
@@ -140,3 +144,126 @@ def chage(self, *args) -> ProcessResult:
         self.host.discard_file("/etc/shadow")
 
         return cmd
+
+    def newusers(self, *args, users_data: str = None) -> ProcessResult:
+        """
+        Update or create new users in batch.
+
+        Updates or creates multiple users by reading account information in passwd
+        format. If `users_data` is provided, it's passed via stdin and takes
+        precedence; otherwise, the command reads from a file specified in `args`.
+        """
+        if users_data:
+            cmd_args = " ".join(args)
+            self.logger.info(f'Creating users from stdin on {self.host.hostname}')
+            cmd = self.host.conn.run(f"echo '{users_data}' | newusers {cmd_args}", log_level=ProcessLogLevel.Error)
+        else:
+            args_dict = self._parse_args(args)
+            self.logger.info(f'Creating users from "{args_dict["name"]}" on {self.host.hostname}')
+            cmd = self.host.conn.run("newusers " + args[0], log_level=ProcessLogLevel.Error)
+
+        self.host.discard_file("/etc/passwd")
+        self.host.discard_file("/etc/shadow")
+        self.host.discard_file("/etc/group")
+        self.host.discard_file("/etc/gshadow")
+
+        return cmd
+
+    def groupmems(self, *args, user: str = "root", password: str = None) -> ProcessResult:
+        """
+        Administer members of a user's primary group.
+
+        The groupmems command allows management of group membership lists.
+        If `user` is provided, then the `password` must be provided and the command groupmems run under this user.
+        If `user` isn't provided, then the command is run under `root`.
+        """
+        args_dict = self._parse_args(args)
+
+        if user == "root":
+            self.logger.info(f'Administer {args_dict["name"]} group membership as root on {self.host.hostname}')
+            cmd = self.host.conn.run("groupmems " + args[0], log_level=ProcessLogLevel.Error)
+        else:
+            self.logger.info(f'Administer {args_dict["name"]} group membership on {self.host.hostname}')
+            cmd = self.host.conn.run(f"echo '{password}' | su - {user} -c 'groupmems {args[0]}'", log_level=ProcessLogLevel.Error)
+
+        self.host.discard_file("/etc/group")
+        self.host.discard_file("/etc/gshadow")
+
+        return cmd
+
+    def newgrp(self, *args, user: str = "root") -> Tuple[ProcessResult, int]:
```

Comment:
Same thing, like newusers, before I knew this was named after the command, a good name would be ```group_tty_mod``` or ```group_ssh_mod```. Just a thought.